crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -352,6 +359,28 @@ impl Backend {
 
     /// Resets the fork to a fresh state
     pub async fn reset_fork(&self, forking: Forking) -> Result<(), BlockchainError> {
+        if !self.is_fork() {
+            if let Some(eth_rpc_url) = forking.clone().json_rpc_url {
+                let mut env = self.env.read().clone();
+
+                let mut node_config = self.node_config.write().await;
+                let (db, forking) =
+                    node_config.setup_fork_db(eth_rpc_url, &mut env, &self.fees).await;
+
+                // TODO: Something like this is needed but...
+                // This won't compile because dyn Db is not Sized (and AFAIK cannot be made Sized)
+                // *self.db.write().await = *db.read().await;
```

Comment:
Would appreciate some help on this part. Really not sure how best to replace the `db` member here. The only thing I can think of now is making the `backend` mutable in `EthApi` and replacing the whole `backend` member there. Maybe that is a better way?